cpp/tests/iterator/value_iterator_test.cuh

```diff
@@ -12,7 +12,9 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-#include "cudf/detail/utilities/vector_factories.hpp"
+#pragma once
+
+#include <cudf/detail/utilities/vector_factories.hpp>
```

Comment:
Should we separate the header groups here? Between `<cudf/detail/utilities/vector_factories.hpp>` and `<tests/iterator/iterator_tests.cuh>`

Reply:
Yes. Since there is only one file per group here, I feel, it's not mandatory. Nonetheless, It's good to separate it.
https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/docs/DEVELOPER_GUIDE.md#includes

Reply:
Another change here is `Order the groups from "nearest" to "farthest".`